sparkles/core.py

```diff
@@ -46,6 +46,34 @@
 CREEP_AWAY_THRESHOLD = 5.0
 
 
+def get_t_ccds_bonus(mags, t_ccd, dyn_bgd_n_faint, dyn_bgd_dt_ccd):
+    """Return array of t_ccds with dynamic background bonus applied.
+
+    This adds ``dyn_bgd_dt_ccd`` to the effective CCD temperature for the
+    ``dyn_bgd_n_faint`` faintest stars, ensuring that at least MIN_DYN_BGD_ANCHOR_STARS
+    are evaluated without the bonus. See:
+    https://nbviewer.org/urls/cxc.harvard.edu/mta/ASPECT/ipynb/misc/guide-count-dyn-bgd.ipynb
+
+    :param mags: array of star magnitudes
+    :param t_ccd: single T_ccd value (degC)
+    :param dyn_bgd_n_faint: number of faintest stars to apply bonus
+    :returns: array of t_ccds (matching ``mags``) with dynamic background bonus applied
+    """
+    t_ccds = np.full_like(mags, t_ccd)
+
+    # If no bonus stars then just return the input t_ccd broadcast to all stars
+    if dyn_bgd_n_faint == 0:
+        return t_ccds
+
+    idxs = np.argsort(mags)
+    n_faint = min(dyn_bgd_n_faint, len(t_ccds))
+    idx_bonus = max(len(t_ccds) - n_faint, MIN_DYN_BGD_ANCHOR_STARS)
+    for idx in idxs[idx_bonus:]:
+        t_ccds[idx] += dyn_bgd_dt_ccd
```

Comment:
Why did this logic switch to using a for loop?  does the previous syntax not work in this case (I don't understand numpy well enough to know)?

`t_ccds[idxs[idx_bonus:]] += dyn_bgd_dt_ccd`

Reply:
I find the loop more readable once we get to needing to nest the index.

Reply:
My first iteration was exactly like the @jskrist version, but I also thought it was less readable than a loop. I also noted that corner cases like where `idx_bonus` is beyond the end of the array might need testing when done as fancy indexing instead of the plain loop. (I tested, it works...).